src/com/google/cose/Ec2SigningKey.java

```diff
@@ -46,31 +46,33 @@ public Ec2SigningKey(DataItem cborKey) throws CoseException, CborException {
     super(cborKey);
 
     keyPair = getKeyPairFromCbor();
-    if ((operations == null)
-        || (operations.contains(Headers.KEY_OPERATIONS_VERIFY)
-        && operations.contains(Headers.KEY_OPERATIONS_SIGN))) {
-      return;
+
+    if ((operations != null)
+        && !operations.contains(Headers.KEY_OPERATIONS_VERIFY)
+        && !operations.contains(Headers.KEY_OPERATIONS_SIGN)) {
+      throw new CoseException("Signing key requires either sign or verify operation.");
     }
-    throw new CoseException("Signing key requires sign and verify operations.");
   }
 
   static class Builder {
     private String keyId;
     private Algorithm algorithm;
-    private final List<Integer> operations;
```

Comment:
This means that a round-trip will reorder the operations.

I'm not sure that's necessarily a bad thing, but I have a hunch it could come back to bite us. IMO, a round-trip from CBOR -> object -> CBOR should product _identical_ CBOR on the output.

Reply:
parse() and serialize() do not use Set at all. So I am not worried about the round trip. Although the builder enforces the operations as a set which may create some problems. But we should let the builder maintain the order and uniqueness of the operations since anyone using builder is depending on us to prevent any error in key generation.

Reply:
Yeah, this is my mistake. The only `Set` types we have to worry about are on the COSE types themselves, not hte builders.